core/src/main/java/org/bitcoinj/core/LegacyAddress.java

```diff
@@ -157,38 +157,31 @@ public static LegacyAddress fromScriptHash(Network network, byte[] hash160) thro
     @Deprecated
     public static LegacyAddress fromBase58(@Nullable NetworkParameters params, String base58)
             throws AddressFormatException, AddressFormatException.WrongNetwork {
-        return fromBase58( (params != null) ? params.network() : null, base58);
+        AddressParser parser = DefaultAddressParser.fromNetworks();
+        return (LegacyAddress) ((params != null)
+                ? parser.parseAddress(base58, params.network())
+                : parser.parseAddressAnyNetwork(base58));
     }
 
     /**
      * Construct a {@link LegacyAddress} from its base58 form.
      *
-     * @param network expected network this address is valid for, or null if the network should be derived from the base58
+     * @param network expected network this address is valid for
      * @param base58 base58-encoded textual form of the address
      * @throws AddressFormatException if the given base58 doesn't parse or the checksum is invalid
      * @throws AddressFormatException.WrongNetwork if the given address is valid but for a different chain (eg testnet vs mainnet)
      */
-    public static LegacyAddress fromBase58(@Nullable Network network, String base58)
+    public static LegacyAddress fromBase58(@Nonnull Network network, String base58)
```

Comment:
I don't think so because the main focus of this commit is that we are migrating the code that handles the "null"/"any" case to `DefaultAddressParser`.

Reply:
Or, that's fine with me.